crates/bevy_core_pipeline/src/dof/dof.wgsl

```diff
@@ -0,0 +1,301 @@
+@group(1) @binding(0) var<uniform> dof_params: DepthOfFieldParams;
+
+// The sampler that's used to fetch texels from the source color buffer.
+@group(1) @binding(1) var color_texture_sampler: sampler;
+
+// cos(-30°), used for the bokeh blur.
+const COS_NEG_FRAC_PI_6: f32 = 0.8660254037844387;
+// sin(-30°), used for the bokeh blur.
+const SIN_NEG_FRAC_PI_6: f32 = -0.5;
+// cos(-150°), used for the bokeh blur.
+const COS_NEG_FRAC_PI_5_6: f32 = -0.8660254037844387;
+// sin(-150°), used for the bokeh blur.
+const SIN_NEG_FRAC_PI_5_6: f32 = -0.5;
+
+// Calculates and returns the diameter (not the radius) of the [circle of
+// confusion].
+//
+// [circle of confusion]: https://en.wikipedia.org/wiki/Circle_of_confusion
+fn calculate_circle_of_confusion(in_frag_coord: vec4<f32>) -> f32 {
+    // Unpack the depth of field parameters.
+    let focus = dof_params.focal_distance;
+    let f = dof_params.focal_length;
+    let scale = dof_params.coc_scale_factor;
+    let max_coc_diameter = dof_params.max_circle_of_confusion_diameter;
+
+    // Sample the depth.
+    let frag_coord = vec2<i32>(floor(in_frag_coord.xy));
+    let raw_depth = textureLoad(depth_texture, frag_coord, 0);
+    let depth = min(-depth_ndc_to_view_z(raw_depth), dof_params.max_depth);
+
+    // Calculate the circle of confusion.
+    //
+    // This is just the formula from Wikipedia [1].
+    //
+    // [1]: https://en.wikipedia.org/wiki/Circle_of_confusion#Determining_a_circle_of_confusion_diameter_from_the_object_field
+    let candidate_coc = scale * abs(depth - focus) / (depth * (focus - f));
+
+    let framebuffer_size = vec2<f32>(textureDimensions(color_texture_a));
+    return clamp(candidate_coc * framebuffer_size.y, 0.0, max_coc_diameter);
+}
+
+// Performs a single direction of the separable Gaussian blur kernel.
+//
+// * `frag_coord` is the screen-space pixel coordinate of the fragment (i.e. the
+//   `position` input to the fragment).
+//
+// * `coc` is the diameter (not the radius) of the circle of confusion for this
+//   fragment.
+//
+// * `frag_offset` is the vector, in screen-space units, from one sample to the
+//   next. For a horizontal blur this will be `vec2(1.0, 0.0)`; for a vertical
+//   blur this will be `vec2(0.0, 1.0)`.
+//
+// Returns the resulting color of the fragment.
+fn gaussian_blur(frag_coord: vec4<f32>, coc: f32, frag_offset: vec2<f32>) -> vec4<f32> {
+    // Usually σ (the standard deviation) is half the radius, and the radius is
+    // half the CoC. So we multiply by 0.25.
+    let sigma = coc * 0.25;
+
+    // 1.5σ is a good, somewhat aggressive default for support—the number of
+    // texels on each side of the center that we process.
+    let support = i32(ceil(sigma * 1.5));
+    let uv = frag_coord.xy / vec2<f32>(textureDimensions(color_texture_a));
+    let offset = frag_offset / vec2<f32>(textureDimensions(color_texture_a));
+
+    // The PDF of the Gaussian blur is (up to constant factors) `exp(-1 / 2σ² *
```

Comment:
Had to google what PDF means, is it the Probability Density Function?
Might be better to spell it out

Reply:
```suggestion
    // The probability density function of the Gaussian blur is (up to constant factors) `exp(-1 / 2σ² *
```